substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/PredicatedConfigurationSupport.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2021, 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package com.oracle.svm.core;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.graalvm.nativeimage.Platform;
+import org.graalvm.nativeimage.Platforms;
+import org.graalvm.nativeimage.hosted.Feature;
+import org.graalvm.nativeimage.impl.ConfigurationPredicate;
+
+@Platforms(Platform.HOSTED_ONLY.class)
+public abstract class PredicatedConfigurationSupport {
```

Comment:
I believe it would be more fitting to implement `ReflectionRegistry` here (all subclasses do) and name this class `AbstractPredicatedReflectionRegistry`.

Reply:
Don't think this is correct. This class will be used later for `com.oracle.svm.reflect.serialize.SerializationSupport`, resources, etc., and those don't implement the reflection registry.

Reply:
I see. I would still prefer `PredicatedConfigurationRegistry` over `*Support`, which we often use in the context of a utility class or a facade class rather than for an abstract base class. The method at the core of this class is called `registerPredicated`, too.

Reply:
Done.